### PR TITLE
refactor(Wrapper) Use arrow functions instead of manual bind

### PR DESCRIPTION
**Pull request by zabute**
I refactored the Wrapper to use arrow functions instead of  binding manually as suggested in https://github.com/formsy/formsy-react/pull/22#issuecomment-345279191 I really think this (or #22) should be released asap as the Wrapper is not usable right now.

Oh and thanks for all the work you're doing to get this project back into shape. :+1:

### CODE REVIEW

File: src/Wrapper.js [hunk 1]
```diff
@@ -116,21 +100,25 @@ export default (Component) => {
...
+      if (!this.isValid() || this.showRequired()) {
+        return this.state.externalError || this.state.validationError || [];
+      }
+
+      return [];
 
-    getValue() {
-      return this.state.value;
+      // !this.isValid() || this.showRequired() ?
+      // (this.state.externalError || this.state.validationError || []) : [];
```
Comment (rkuykendall) [inline src/Wrapper.js]: ?

File: .eslintrc [hunk 2]
```diff
@@ -1,3 +1,4 @@
 {
+  "parser": "babel-eslint",
```
Comment (rkuykendall) [inline .eslintrc]: 👌

File: src/Wrapper.js [hunk 3]
```diff
@@ -154,36 +142,23 @@ export default (Component) => {
...
-    isValid() {
-      return this.state.isValid;
-    }
+    isValid = () => this.state.isValid;
 
-    isValidValue(value) {
-      return this.context.formsy.isValidValue.call(null, this, value);
+    isValidValue = value =>
+      this.context.formsy.isValidValue.call(null, this, value);
       // return this.props.isValidValue.call(null, this, value);
```
Comment (rkuykendall) [inline src/Wrapper.js]: I may need to do a sweep of the codebase for commented out code...

File: src/Wrapper.js (hunk 1 above)
Comment (zabute) [reply src/Wrapper.js]: eslint was complaining about ambiguous usage of ternary operators.

Comment (rkuykendall) [reply src/Wrapper.js]: Ah, I see. Yeah the older code was definitely confusing. This is MUCH cleaner. Thanks! I would delete the commented-out older version and the blank line before the closing bracket. No need to keep it around.

File: src/Wrapper.js (hunk 3 above)
Comment (zabute) [reply src/Wrapper.js]: Shall I clean them up in a separate PR?

Comment (rkuykendall) [reply src/Wrapper.js]: Would you mind removing this commented out line? I don't see any reason we would keep it around.

File: src/Wrapper.js [hunk 4]
```diff
@@ -154,36 +142,23 @@ export default (Component) => {
...
 
-    hasValue() {
-      return this.state.value !== '';
-    }
+    hasValue = () => this.state.value !== '';
 
-    isFormDisabled() {
-      return this.context.formsy.isFormDisabled();
-    }
+    isFormDisabled = () => this.context.formsy.isFormDisabled();
```
Comment (rkuykendall) [inline src/Wrapper.js]: These are really nice. Loved the simplified form.

File: src/Wrapper.js (hunk 3 above)
Comment (rkuykendall) [reply src/Wrapper.js]: @zabute I actually searched around and couldn't find any others imminently. If you see more, then yeah, we can just leave them for a future PR.

**Comment by rkuykendall** [comment]
This looks good to me! Thanks for all your help!

**Pull request merged**